src/lib.rs

```diff
@@ -1 +1,203 @@
+                directory,
+                &startup_info,
+                &mut process_info as *mut PROCESS_INFORMATION,
+            )
+        } else {
+            windows::Win32::System::Threading::CreateProcessW(
+                PWSTR::default(),
+                command,
+                std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                inherit_handles,
+                process_creation_flags,
+                std::ptr::null() as *const c_void,
+                PWSTR::default(),
+                &startup_info,
+                &mut process_info as *mut PROCESS_INFORMATION,
+            )
+        };
+
+        if res.as_bool() {
+            Ok(Self {
+                process_information: process_info,
+            })
+        } else {
+            Err(Error::CreationFailed(GetLastError().0))
+        }
+    }
+
+    pub fn wait(&self) -> Result<ExitStatus> {
+        let mut exit_code: u32 = 0;
+
+        unsafe {
+            let res = WaitForSingleObject(self.process_information.hProcess, INFINITE);
+
+            if res == WAIT_OBJECT_0 {
+                if GetExitCodeProcess(
+                    self.process_information.hProcess,
+                    &mut exit_code as *mut u32,
+                )
+                .as_bool()
+                {
+                    close_handle(&self.process_information);
+                    Ok(ExitStatus(exit_code))
+                } else {
+                    Err(Error::GetExitCodeFailed(GetLastError().0))
+                }
+            } else {
+                Err(Error::GetExitCodeFailed(GetLastError().0))
+            }
+        }
+    }
+
+    pub fn try_wait(&self) -> Result<Option<ExitStatus>> {
+        let mut exit_code: u32 = 0;
+
+        let res = unsafe {
+            GetExitCodeProcess(
+                self.process_information.hProcess,
+                &mut exit_code as *mut u32,
+            )
+        };
+
+        if res.as_bool() {
+            if exit_code == STATUS_PENDING.0 {
+                Ok(None)
+            } else {
+                close_handle(&self.process_information);
+                Ok(Some(ExitStatus(exit_code)))
+            }
+        } else {
+            Err(Error::GetExitCodeFailed(unsafe { GetLastError().0 }))
+        }
+    }
+
+    pub fn kill(&self) -> Result<()> {
+        let res = unsafe { TerminateProcess(self.process_information.hProcess, 0) };
+
+        if res.as_bool() {
+            Ok(())
+        } else {
+            Err(Error::KillFailed(unsafe { GetLastError().0 }))
+        }
+    }
+}
+
+pub struct ExitStatus(u32);
+
+impl ExitStatus {
+    pub fn success(&self) -> bool {
+        self.0 == 0
+    }
+
+    pub fn code(&self) -> u32 {
+        self.0
+    }
+}
+
+fn close_handle(process_info: &PROCESS_INFORMATION) {
+    unsafe {
+        CloseHandle(process_info.hProcess);
+        CloseHandle(process_info.hThread);
+    }
+}
```

Comment:
Don't wrap unsafe function into safe function pretending it is okay. You need to know for sure that it is okay.

Here the fact that you take a lower level object in argument makes the entire function unsafe.

```suggestion
unsafe fn close_handle(process_info: &PROCESS_INFORMATION) {
    CloseHandle(process_info.hProcess);
    CloseHandle(process_info.hThread);
}
```